docs/migration-guide/w3c-vc-v2.md

```diff
@@ -0,0 +1,565 @@
+// Add credential subject
+document.credentialSubject({
+  id: 'did:example:student123',
+  name: 'Alice Johnson',
+  studentId: 'STU-2024-001',
+  degree: 'Bachelor of Computer Science',
+  gpa: 3.9,
+  graduationYear: 2024,
+  dateOfBirth: '1999-05-15'
+});
+
+// Set validUntil (v2.0 date fields)
+document.expirationDate('2029-01-01T00:00:00Z');
+
+// Add credential status for revocation (v2.0 format) - Verifiable Document
+document.credentialStatus({
+  id: 'https://example.com/status/1#42',
+  type: 'BitstringStatusListEntry',
+  statusPurpose: 'revocation',
+  statusListIndex: '42',
+  statusListCredential: 'https://example.com/status/1'
+});
+
+// Or add credential status for Transferable Records
+document.credentialStatus({
+  chain: 'Ethereum',
+  chainId: 1,
+  tokenRegistry: '0x1234567890abcdef...',
+  rpcProviderUrl: 'https://mainnet.infura.io/v3/YOUR_INFURA_PROJECT_ID',
+});
+```
+
+#### Document Builder with Signing and Selective Disclosure
+
+```typescript
+// ECDSA-SD-2023 key pair
+const modernKeyPair = {
+  '@context': 'https://w3id.org/security/multikey/v1',
+  id: 'did:web:example.com#multikey-1',
+  controller: 'did:web:example.com',
+  type: 'Multikey',
+  secretKeyMultibase: 'z...',
+  publicKeyMultibase: 'z...'
+};
+
+// Sign with mandatory pointers (always visible fields)
+const signedDocument = await document.sign(modernKeyPair, 'ecdsa-sd-2023', {
+  mandatoryPointers: [
+    '/credentialSubject/id',
+    '/credentialSubject/name',
+    '/credentialSubject/degree'
+  ]
+});
+console.log(signedDocument);
+```
+
+> **Mandatory Pointers**: In ECDSA-SD-2023, mandatory pointers specify which fields must always be disclosed and cannot be selectively hidden. This is useful for ensuring critical information like the credential subject's ID and name are always visible, while other fields like age or GPA can be selectively disclosed. The pointers use JSON Pointer syntax (RFC 6901) to reference specific fields in the credential.
+> 
+> **Default Behavior**: If no mandatory pointers are provided, all fields in the credential become selectively disclosable, meaning the holder can choose to hide any field during presentation. However, certain fields like the credential's `@context`, `type`, `issuer`, and `validFrom`/`validUntil` are typically always disclosed by default as they are essential for credential verification.
+
+#### Built-in Derive Function
+
+The new DocumentBuilder includes a built-in derive function for selective disclosure:
+
+```typescript
+// Use DocumentBuilder's derive function for selective disclosure
+const derivedDocument = await document.derive([
+  '/credentialSubject/graduationYear'
+  // Reveals: id, name, degree (mandatory) + graduationYear
+  // Hides: studentId, gpa, dateOfBirth
+]);
+console.log(derivedDocument);
+
+// Verify the derived credential
+// ECDSA-SD-2023 credentials must be derived before verification
+const isVerified = await document.verify();
+console.log(isVerified); // true or false
+```
```

Comment:
can we derive before signing? if cannot then can we put derive function at the back, behind advanced document builder features, last function?